frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/api/TopicResource.java

```diff
@@ -0,0 +1,169 @@
+package com.mesosphere.sdk.kafka.api;
+
+import com.mesosphere.sdk.kafka.cmd.CmdExecutor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Topic Resource executing commands through command executor.
+ * Kafka package should also be deployed with framework to enable Kafka commands.
+ */
+
+@Path("/v1/topics")
+public class TopicResource {
+    private static final Log log = LogFactory.getLog(TopicResource.class);
+
+    private final KafkaZKClient kafkaZkClient;
+    private final CmdExecutor cmdExecutor;
+
+    public TopicResource(CmdExecutor cmdExecutor, KafkaZKClient kafkaZkClient) {
+        this.kafkaZkClient = kafkaZkClient;
+        this.cmdExecutor = cmdExecutor;
+    }
+
+    @GET
+    public Response topics() {
+        try {
+            return Response.ok(kafkaZkClient.listTopics(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/{name}")
+    public Response getTopic(@PathParam("name") String topicName) {
+        try {
+            return Response.ok(kafkaZkClient.getTopic(topicName), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topic: " + topicName + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @POST
+    @Path("/create")
```

Comment:
Alternatively, since we index by name, that means you know the resource's path beforehand (e.g. `/v1/topics/{name}`). In that case, it would also be acceptable to have this method be a `PUT` to `/v1/topics/{name}`. Either way seems fine.